lib/europeana/mixins/image_version.rb

```diff
@@ -31,16 +31,18 @@ def versions(name = 'image')
           image = @element.content_by_name(name)
           if image&.essence&.picture.present?
             picture = image.essence.picture
-            VERSIONS.each_with_object({}) do |(version, settings), memo|
-              alchemy_picture_version = picture_version_from_key(picture, version) || picture_version(picture, settings)
-              url = picture_version_url(alchemy_picture_version)
-              memo[version] = { url: url }
-            end
+            version_uuid_pairs(picture.id).map { |key, uuid| [key.to_sym, { url: picture_version_url(uuid) }] }.to_h
           else
             false
           end
         end
       end
+
+      def version_uuid_pairs(picture_id)
+        signature_criteria = { picture_id: picture_id, version_key: VERSIONS.keys }
```

Comment:
Why is `version_key` needed as a criterion? When would it ever not be in `VERSION.keys`?

Reply:
Currently we always use all the versions. The only reason this is specified now is in case we change the versions we use, like for example when we removed the halfx2 etc. versions. Theoretically we wouldn't need to delete the versions since we can retrieve only what's needed. 

Also there's a possibility we may want to extend this method in future to selectively retrieve versions. For example we should only need to retrieve facebook/twitter versions for one image per page. However we don't have a good way of distinguishing which versions will be needed currently.